### ISSUE TITLE
Filter manyToMany relation with users-permission plugin

### ISSUE THREAD

**Issue by plduhoux** [strapi/strapi#3200]
**Informations**
- **Node.js version**: 10.15.1
- **NPM version**: 6.9.0
- **Strapi version**: 3.0.0-alpha.26.1 
- **Database**: MySQL
- **Operating system**: macOS

**What is the current behavior?**
I am trying to use deep filtering for a relation between my model and users permission plugin (manyToMany). Here is the users attribute on the model :
```
    "users": {
      "collection": "user",
      "via": "campaigns",
      "dominant": true,
      "plugin": "users-permissions"
    }
```
The intermediate table generated by strapi (i am using bookshelf and mysql) contains the following columns : id, **user_id**, campaign_ id
When I try to filter campaign using user_id (with parameter users_in=x in the fetchAll params argument), the generated query fails, here is the query :
```
select distinct count(*) as `count` from `campaigns` left join `db`.`campaigns_users__users_campaigns` as `campaigns_users__users_campaigns_1` on `campaigns_users__users_campaigns_1`.`campaign_id` = `campaigns`.`id` left join `db`.`users-permissions_user` as `users-permissions_user_1` on `campaigns_users__users_campaigns_1`.`userspermissionsuser_id` = `users-permissions_user_1`.`id` where `users-permissions_user_1`.`id` in (2)
```
It is assuming that the intermediate table between campaigns and users contains a userspermissionsuser_id column...
Here is the error :
```
code: 'ER_BAD_FIELD_ERROR',
  errno: 1054,
  sqlMessage:
   'Unknown column \'campaigns_users__users_campaigns_1.userspermissionsuser_id\' in \'on clause\'',
  sqlState: '42S22',
  index: 0,
```
If I rename the **userspermissionsuser_id** column to **user_id**, the request works fine, but then if I insert a new campaign with a related user, the **user_id** column is recreated by strapi and filled with the concerned user, **userspermissionsuser_id** is null

TL;DR: When using manyToMany with users plugin, strapi is generating an intermediate table containing column **user_id**, deep filtering asssumes that this very same column is named **userspermissionsuser_id**

**Steps to reproduce the problem**
Create a content type with a manyToMany relation to users-permission plugin. (using a relational database > bookshelf)

Query this content type using params "users_in=x", the request fails because the expected column name in the intermediate table is wrong

**What is the expected behavior?**
Strapi should expect the column name in the intermediate table for a mtm relation to be the one it generated

**Comment by seegan** [comment]
I have got the same issue too 


"message": "select distinct `users-permissions_user`.* from `users-permissions_user` left join `qpunch`.`tasks_users__users_tasks` as `tasks_users__users_tasks_1` on `tasks_users__users_tasks_1`.`userspermissionsuser_id` = `users-permissions_user`.`id` left join `qpunch`.`tasks` as `tasks_1` on `tasks_users__users_tasks_1`.`task_id` = `tasks_1`.`id` where `tasks_1`.`id` in ('2') - ER_BAD_FIELD_ERROR: Unknown column 'tasks_users__users_tasks_1.userspermissionsuser_id' in 'on clause'",

**Comment by seegan** [comment]
I just workaround with the node_modules/strapi-hook-bookshelf/lib/buildQuery.js Line number 85 by

```
      qb.leftJoin(
        `${originInfo.model.databaseName}.${assoc.tableCollectionName} AS ${joinTableAlias}`,
        `${joinTableAlias}.${singular(originInfo.model.collectionName)}_${
          originInfo.model.attributes[assoc.alias].column
        }`,
        `${originInfo.alias}.${originInfo.model.primaryKey}`
      );


to 

      qb.leftJoin(
        `${originInfo.model.databaseName}.${assoc.tableCollectionName} AS ${joinTableAlias}`,
        `${joinTableAlias}.${
          originInfo.model.attributes[assoc.alias].column
        }`,
        `${originInfo.alias}.${originInfo.model.primaryKey}`
      );
```
Sorry Still I don't know whether it is right or wrong, but I have to show my project to client tomorrow..

**Comment by lauriejim** [comment]
This issue is fixed in the `beta` version.
Will be release before the end of the month.
